src/pudl/transform/ferc1.py

```diff
@@ -891,6 +883,68 @@ def reconcile_table_calculations(
     return calculated_df
 
 
+def calculate_values_from_components(
+    calculation_components: pd.DataFrame,
+    data: pd.DataFrame,
+    validate: Literal["one_to_many", "many_to_many"],
+    calc_idx: list[str],
+    data_idx: list[str],
```

Comment:
check to see if i can always just make `data_idx` by `calc_idx + ["utility_id", "report_year"]`

Reply:
done!